libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/DecrypterInputStream.java

```diff
@@ -45,7 +45,12 @@
 
     public DecrypterInputStream(FileInputStream inputStream, String encryptionKey)
             throws GeneralSecurityException, IOException {
+        // First byte should be iv length
         int ivLength = inputStream.read();
+        if (ivLength != 16 && ivLength != 32) {
```

Comment:
Without that check it would fail later (probably in the getDecryptingCipher method) but the error could be hard to make sense of.